From 5c31b67c43f2d6c638b6ac7f2b2a8cfffe1d0c11 Mon Sep 17 00:00:00 2001
From: Simon McVittie <simon.mcvittie@collabora.co.uk>
Date: Tue, 25 Oct 2016 19:43:32 +0100
Subject: [PATCH 4/6] virt-qemu: re-create eofcat on every boot

This is currently useless, but will become useful when I put it
on a tmpfs.

Signed-off-by: Simon McVittie <simon.mcvittie@collabora.co.uk>
---
 virt/autopkgtest-virt-qemu | 70 +++++++++++++++++++++++-----------------------
 1 file changed, 35 insertions(+), 35 deletions(-)

diff --git a/virt/autopkgtest-virt-qemu b/virt/autopkgtest-virt-qemu
index d9c59a2..24e4353 100755
--- a/virt/autopkgtest-virt-qemu
+++ b/virt/autopkgtest-virt-qemu
@@ -234,41 +234,6 @@ touch /run/autopkgtest/shared/done_shared
     term.send(b'[ -n "$HOME" ] || export HOME=`getent passwd root|cut -f6 -d:`\n')
     VirtSubproc.expect(term, b'#', 5)
 
-
-def setup_config(shared_dir):
-    '''Set up configuration files'''
-
-    term = VirtSubproc.get_unix_socket(os.path.join(workdir, 'ttyS1'))
-
-    # copy our timezone, to avoid time skews with the host
-    if os.path.exists('/etc/timezone'):
-        tz = None
-        with open('/etc/timezone', 'rb') as f:
-            for l in f:
-                if l.startswith(b'#'):
-                    continue
-                l = l.strip()
-                if l:
-                    tz = l
-                    break
-
-        if tz:
-            adtlog.debug('Copying host timezone %s to VM' % tz.decode())
-            term.send(b'echo ' + tz + b' > /etc/timezone; DEBIAN_FRONTEND=noninteractive dpkg-reconfigure tzdata\n')
-            VirtSubproc.expect(term, b'#', 30)
-        else:
-            adtlog.debug('Could not determine host timezone')
-
-    # ensure that we have Python for our the auxverb helpers
-    term.send(b'type python3 2>/dev/null || type python 2>/dev/null\n')
-    try:
-        out = VirtSubproc.expect(term, b'/python', 5)
-    except VirtSubproc.Timeout:
-        VirtSubproc.bomb('Neither python3 nor python is installed in the VM, '
-                         'one of them is required by autopkgtest')
-    if b'\n# ' not in out:
-        VirtSubproc.expect(term, b'# ', 5)
-
     # create helper for runcmd: cat data from its stdin (from a file) to stdout
     # eternally (like tail -f), but stop once either an "EOF" file exists and
     # we copied at least as many bytes as given in that EOF file (the first
@@ -308,6 +273,41 @@ EOF
     VirtSubproc.expect(term, b'# ', 5)
 
 
+def setup_config(shared_dir):
+    '''Set up configuration files'''
+
+    term = VirtSubproc.get_unix_socket(os.path.join(workdir, 'ttyS1'))
+
+    # copy our timezone, to avoid time skews with the host
+    if os.path.exists('/etc/timezone'):
+        tz = None
+        with open('/etc/timezone', 'rb') as f:
+            for l in f:
+                if l.startswith(b'#'):
+                    continue
+                l = l.strip()
+                if l:
+                    tz = l
+                    break
+
+        if tz:
+            adtlog.debug('Copying host timezone %s to VM' % tz.decode())
+            term.send(b'echo ' + tz + b' > /etc/timezone; DEBIAN_FRONTEND=noninteractive dpkg-reconfigure tzdata\n')
+            VirtSubproc.expect(term, b'#', 30)
+        else:
+            adtlog.debug('Could not determine host timezone')
+
+    # ensure that we have Python for our the auxverb helpers
+    term.send(b'type python3 2>/dev/null || type python 2>/dev/null\n')
+    try:
+        out = VirtSubproc.expect(term, b'/python', 5)
+    except VirtSubproc.Timeout:
+        VirtSubproc.bomb('Neither python3 nor python is installed in the VM, '
+                         'one of them is required by autopkgtest')
+    if b'\n# ' not in out:
+        VirtSubproc.expect(term, b'# ', 5)
+
+
 def make_auxverb(shared_dir):
     '''Create auxverb script'''
 
-- 
2.10.1

